src/runtime/graph/graph_runtime.cc

```diff
@@ -103,27 +103,56 @@ class GraphRuntime : public ModuleNode {
   void SetInput(int index, DLTensor* data_in) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(data_in, &data_entry_[eid], nullptr));
+    data_entry_[eid].CopyFrom(data_in);
   }
   /*!
-   * \brief Copy index-th input to data_out
+   * \brief Get the number of outputs
+   *
+   * \return The number of outputs from graph.
+   */
+  int NumOutputs() {
+    return outputs_.size();
+  }
+  /*!
+   * \brief Return NDArray for given input index.
    * \param index The input index.
-   * \param data_out The output
+   *
+   * \return NDArray corresponding to given input node index.
    */
-  void GetInput(int index, DLTensor* data_out) {
+  NDArray GetInput(int index) {
     CHECK_LT(static_cast<size_t>(index), input_nodes_.size());
     uint32_t eid = this->entry_id(input_nodes_[index], 0);
-    TVM_CCALL(TVMArrayCopyFromTo(&data_entry_[eid], data_out, nullptr));
+    return data_entry_[eid];
+  }
+  /*!
+   * \brief Return NDArray for given output index.
+   * \param index The output index.
+   *
+   * \return NDArray corresponding to given output node index.
+   */
+  NDArray GetOutput(int index) {
+    CHECK_LT(static_cast<size_t>(index), outputs_.size());
+    uint32_t eid = this->entry_id(outputs_[index]);
+    tvm::runtime::DeviceAPI::Get(ctx_)->StreamSync(ctx_, nullptr);
```

Comment:
Should not call streamSync here, and directly return the output, caller is expected to call sync, after copy(NOTE that sync will be automatically in effect when asnumpy is called)